reconcile/openshift_upgrade_watcher.py

```diff
@@ -1,39 +1,51 @@
 import logging
 
 from datetime import datetime
+from typing import Optional
 
 from reconcile import queries
 
 from reconcile.slack_base import slackapi_from_queries
 from reconcile.utils.oc import OC_Map
+import reconcile.utils.ocm as ocmmod
+from reconcile.utils.slack_api import SlackApi
 from reconcile.utils.state import State
 from reconcile.utils.defer import defer
 
 
 QONTRACT_INTEGRATION = "openshift-upgrade-watcher"
 
 
-@defer
-def run(dry_run, thread_pool_size=10, internal=None, use_jump_host=True, defer=None):
-    settings = queries.get_app_interface_settings()
-    accounts = queries.get_state_aws_accounts()
-    clusters = [c for c in queries.get_clusters(minimal=True) if c.get("ocm")]
-    oc_map = OC_Map(
-        clusters=clusters,
-        integration=QONTRACT_INTEGRATION,
-        settings=settings,
-        internal=internal,
-        use_jump_host=use_jump_host,
-        thread_pool_size=thread_pool_size,
-    )
-    defer(oc_map.cleanup)
-    state = State(
-        integration=QONTRACT_INTEGRATION, accounts=accounts, settings=settings
-    )
+def cluster_slack_handle(cluster: str, slack: Optional[SlackApi]):
+    usergroup = f"{cluster}-cluster"
+    usergroup_id = f"@{usergroup}"
+    if slack:
+        usergroup_id = slack.get_usergroup_id(usergroup)
+    return f"<!subteam^{usergroup_id}>"
+
+
+def slack_notify(
```

Comment:
So this function does more than notify, correct? which makes the name of it quite misleading as I didn't expect it to handle the logic of both the state and the message in one go. 

It is very convenient to have it here as it makes the code very similar in both cases... But I would love a name that describes what it does, something about handling the state and notifying...

Maybe something like this, if you like it

```suggestion
def handle_slack_notification(
    """Check notification status, notify if needed and update the notification status"""
```